### PR TITLE
Added support for the Keebio Levinson Rev 3 Shield.

### PR DESCRIPTION
**Pull request by blistergeist**
Added support for the [Keebio Levinson Rev 3](https://keeb.io/products/levinson-lets-split-w-led-backlight) shield. Tested locally on nRFMicro revision 1.3.

### CODE REVIEW

File: app/boards/shields/levinson/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#Copyright (c) 2021 The ZMK Contributors
+#SPDX-License-Identifier: MIT
```
Comment (Nicell) [inline app/boards/shields/levinson/Kconfig.defconfig:2]:
```suggestion
# Copyright (c) 2021 The ZMK Contributors
# SPDX-License-Identifier: MIT
```

**Review by Nicell — CHANGES_REQUESTED** [review]
Great work on this shield. Super tiny comment and then just these files to update:

[`.github/workflows/build.yml`](https://github.com/zmkfirmware/zmk/blob/main/.github/workflows/build.yml): Add your board to the shield list.

[`docs/docs/hardware.md`](https://github.com/zmkfirmware/zmk/blob/main/docs/docs/hardware.md): You should add the Levinson to this list

[`docs/static/setup.ps1`](https://github.com/zmkfirmware/zmk/blob/main/docs/static/setup.ps1) and [`docs/static/setup.sh`](https://github.com/zmkfirmware/zmk/blob/main/docs/static/setup.sh): Levinson should be added as a split option for both files.

File: app/boards/shields/levinson/Kconfig.defconfig (hunk 1 above)
Comment (blistergeist) [reply app/boards/shields/levinson/Kconfig.defconfig]: @Nicell thanks for the further instructions, I have added the Levinson as an option in the files you mentioned following the formatting of all the other shields. I hope this was correct. I also added in those spaces in the copyright lines. Not sure how that one got missed ;)

File: app/boards/shields/levinson/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	default "Levinson Left"
+
+endif
+
+if SHIELD_LEVINSON_RIGHT
+
+config ZMK_KEYBOARD_NAME
+	default "Levinson Right"
+
+endif
```
Comment (Nicell) [inline app/boards/shields/levinson/Kconfig.defconfig:17]: A recent change was made to how we define the central and peripheral for splits, which can be seen here: https://zmkfirmware.dev/docs/development/new-shield#kconfigdefconfig. Could you update this file to match the docs? Along with this you'll need to clear out the left and right `.conf` files.

File: app/boards/shields/levinson/levinson_left.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CONFIG_ZMK_SPLIT=y
+CONFIG_ZMK_SPLIT_BLE_ROLE_CENTRAL=y
```
Comment (Nicell) [inline app/boards/shields/levinson/levinson_left.conf:2]:
```suggestion
```

File: app/boards/shields/levinson/levinson_right.conf [hunk 4]
```diff
@@ -0,0 +1 @@
+CONFIG_ZMK_SPLIT=y
```
Comment (Nicell) [inline app/boards/shields/levinson/levinson_right.conf:1]:
```suggestion
```

**Review by Nicell — CHANGES_REQUESTED** [review]
I missed this on my first go around! We've changed how split roles are defined, and I've added comments on the files that need to change. Thanks!

File: app/boards/shields/levinson/Kconfig.defconfig (hunk 2 above)
Comment (blistergeist) [reply app/boards/shields/levinson/Kconfig.defconfig]: Changes made and pushed.

**Review by Nicell — COMMENTED** [review]
@blistergeist The changes look good! We're getting #546 merged in first, then once that's merged I'll ask you to rebase this PR hopefully one last time. Then it should be ready for a final review and merge 👍 

I can notify you here when we're ready for you to rebase.

**Comment by blistergeist** [comment]
Great! Looking forward to it.

**Comment by blistergeist** [comment]
@Nicell, status check on this? Looks like #546 is ready pending review.

**Comment by Nicell** [comment]
@blistergeist Sorry about the huge delay here, for some reason our update message didn't get put on this PR, here's a copy of it:

Thank you, contributor, for your patience with how long review and merge of boards/shields has taken!

There are three recent refactors/changes to boards and shields that require some attention, and then we can finally get this PR merged!

1. Hardware Metadata
1. Pro Micro shield DT naming changes
1. Split changes for BLE advertising

# Hardware Metadata

## The Problem

When first developing the process around contributing new shields/boards to ZMK, we failed to recognize that several key files (setup scripts, documentation page of supported hardware, and GH Action `build.yml` file) required changes, often in the same spot, for every PR. This resulted in immediate merge conflicts for every *other* PR after one was merged, which is a  headache for contributors.

## The Fix

By adding discrete metadata files that are located with the boards/shields in question, and using that metadata to generate setup scripts, website hardware list, etc., users can contributing new hardware descriptions without the need to change the same files that other contributors are changing.

## Next Steps

First, refer to https://zmk.dev/docs/development/hardware-metadata-files to familiarize yourself with the new metadata file format.

Next, you have two options for fixing up your PR:

1. If comfortable with `git rebase`, perform an interactive rebase and revert any changes to `build.yml`, `hardware.md`, or the setup scripts `setup.sh`/`setup.ps1`, and then add the new metadata YAML file. Then force push your branch. Or,
2. Create a new branch from an up-to-date `main`, copy in the files for your new hardware, add the metadata file, then commit and push the new branch. Then, edit your open PR to point to your new branch.

# Pro Micro shield DT naming changes

In #876, we have simplified the DT naming for the "nexus node" we expose for pro-micro compatible boards, deprecating the use of `pro_micro_a`, and renaming `pro_micro_d` node to simply `pro_micro`. For pro-micro boards and shields, you'll need to adjust your DT to use the proper names.

Please see https://zmk.dev/docs/development/new-shield#shield-overlays for the updated docs on this.

# Split Shield Advertising Changes

In addition, if this is a split PR, please see #658 where we have changed our conventions to *remove* the the name from the right sides, to prevent users attempting to pair with them and causing split sync issues. This also includes removing the " Left" suffix from the naming on the left side. See the changes in that PR for examples of what to change with your split shield.

# Getting Help

If you have any questions about any of these changes, please comment here and tag @zmkfirmware/boards-shields or ask in the `#boards-shields` Discord channel.

**Comment by Nicell** [comment]
@blistergeist  I'll be closing this PR as we haven't heard anything in 6 months. Please feel free to reopen this if you ever come back to it.

**Pull request closed without merging**